p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -205,24 +213,47 @@ public GetUpdatedDataRequest buildGetUpdatedDataRequest(NodeAddress senderNodeAd
         return new GetUpdatedDataRequest(senderNodeAddress, nonce, this.getKnownPayloadHashes());
     }
 
+    private byte[] getSpecialKey() {
+        byte[] result = new byte[20];
+        Arrays.fill(result, (byte) 0);
+        System.arraycopy(Version.VERSION.getBytes(), 0, result, 0, Version.VERSION.length());
+        return result;
+    }
+
+    private String containsSpecialKey(Set<P2PDataStorage.ByteArray> collection) {
+        Optional<String> result = collection.stream().map(byteArray -> new String(byteArray.bytes).trim()).filter(s -> s.matches("^[0-9]\\.[0-9]\\.[0-9]$")).findFirst();
```

Comment:
Is String(byte[]) safe enough? I suspect so but I'm not sure.

Reply:
at least I haven't found an issue yet

Reply:
I ask since it might act differently on different platforms. I think it would never be an issue for version values, but perhaps other values could cause problems, deterministic is nice.

Reply:
Just played around with it a bit. Doing an old-fashioned for-switch-if routine is about 10 times faster. However, 10 times faster means it takes an average of 2,6ms vs. 23,4ms (60k objects, average over 100 tries) for an operation that is performed twice during the runtime of Bisq (file loading time is about 3 seconds). In contrast, it takes 20 LOC more to achieve this. 

ad deterministic behavior: I haven't found any hint on non-deterministic behavior. Creating a String from byte[] is pretty standard. I, however, added a charset for good measure. Although "\d.\d.\d" contains nothing another charset would handle differently.